modules/contact/src/constraints/ComputeWeightedGapLMMechanicalContact.C

```diff
@@ -162,12 +164,27 @@ ComputeWeightedGapLMMechanicalContact::computeQpIProperties()
                               : static_cast<const DofObject *>(_lower_secondary_elem);
 
   if (_interpolate_normals)
-    _dof_to_weighted_gap[dof].first += _test[_i][_qp] * _qp_gap;
+  {
+    if (_use_petrov_galerkin)
+      _dof_to_weighted_gap[dof].first += _test_aux_lm[_i][_qp] * _qp_gap;
+    else
+      _dof_to_weighted_gap[dof].first += _test[_i][_qp] * _qp_gap;
```

Comment:
You could create a variable to hold the value of `_test` or `test_aux_lm` depending on the user selection. That would eliminate additional branching/lines of code. 

On a separate note, we'd need to figure out how to access regular Lagrange test functions from the user objects per the design that @lindsayad created (https://github.com/idaholab/moose/pull/23591) --but we can figure that out later once we assess how this approach behaves in our target problems.